include/circt/Dialect/RTL/Combinatorial.td

```diff
@@ -120,12 +108,27 @@ def AndOp : UTVariadicRTLOp<"and", [Commutative]>;
 def OrOp  : UTVariadicRTLOp<"or", [Commutative]>;
 def XorOp : UTVariadicRTLOp<"xor", [Commutative]>;
 
-def EQOp   : CmpBinRTLOp<"eq", [Commutative]>;
-def NEQOp  : CmpBinRTLOp<"neq", [Commutative]>;
-def LTOp   : CmpBinRTLOp<"lt">;
-def ULTOp  : CmpBinRTLOp<"ult">;
-def LEQOp  : CmpBinRTLOp<"leq">;
-def ULEQOp : CmpBinRTLOp<"uleq">;
+def ICmpPredicateEQ  : I64EnumAttrCase<"eq", 0>;
+def ICmpPredicateNE  : I64EnumAttrCase<"ne", 1>;
+def ICmpPredicateSLT : I64EnumAttrCase<"slt", 2>;
+def ICmpPredicateSLE : I64EnumAttrCase<"sle", 3>;
+def ICmpPredicateULT : I64EnumAttrCase<"ult", 4>;
+def ICmpPredicateULE : I64EnumAttrCase<"ule", 5>;
+def ICmpPredicate : I64EnumAttr<
+    "ICmpPredicate",
+    "rtl.icmp comparison predicate",
+    [ICmpPredicateEQ, ICmpPredicateNE,
+     ICmpPredicateSLT, ICmpPredicateSLE,
+     ICmpPredicateULT, ICmpPredicateULE]>;
+
+// Other integer operations.
+def ICmpOp : RTLOp<"icmp", [NoSideEffect, SameTypeOperands]> {
+  let arguments = (ins ICmpPredicate:$predicate, AnySignlessInteger:$lhs, AnySignlessInteger:$rhs);
+  let results = (outs I1:$result);
+
+  let parser = [{ return parseICmpOp(parser, result); }];
```

Comment:
I thought the ODS generated asmparser/printer supported enums.  Is this missing?

Reply:
I might have missed it in the docs.  I'll try.